stockflux-securities-master/src/components/InputForm.js

```diff
@@ -27,47 +37,60 @@ const InputForm = ({ match }) => {
 
   useEffect(() => {
     if (match.params.securityId) {
-      setIsLoading(true);
+      setContextOrState(stateEnum.loading);
       getSecurity(match.params.securityId)
         .then(security => {
-          setIsLoading(false);
-          setErrorMessage(null);
+          setContextOrState(null);
           setSecurityState(security);
         })
         .catch(() => {
-          setIsLoading(false);
-          setErrorMessage("Error, cannot get security");
+          setContextOrState(stateEnum.error);
+          setMessages(["Error, cannot get security"]);
         });
     }
   }, []);
 
   const submitForm = event => {
     event.preventDefault();
-    setInProgress(true);
-
+    setContextOrState(stateEnum.sending);
```

Comment:
Don't forget to clear the messages here too.

Reply:
As it's being assigned to either an error message or a success message later in the post security function .then() or .catch() does it need to be cleared beforehand? 

There's no chance of it not being reassigned so I don't think clearing it beforehand will do anything

Reply:
At the point where the form has been submitted, it should be neither in a success nor error state and therefore shouldn't have any messages. Clearing the messages here will have the effect of removing the success/error messages from view until the submit/save completes. It's very minor but it goes to keeping the state more correct at each point in time.

Reply:
Ah cool, I understand. So if the submitting takes a while there will be no message while it'a waiting. I've now made the change.